dts/bindings/sensor/maxim,max30101.yaml

```diff
@@ -1,8 +1,250 @@
+        Default: 2, compromise between saturation and resolution
+    smp-rate-ctrl:
+      type: int
+      required: false
+      default: 0
+      enum:
+       - 0
+       - 1
+       - 2
+       - 3
+       - 4
+       - 5
+       - 6
+       - 7
+      description: |
+        Set the effective sampling rate with one sample consisting of one
+        pulse/conversion per active LED channel. In SpO2 mode, these means
+        one IR pulse/conversion and one red pulse/conversion per sample
+        period.
+        0 = 50 Hz
+        1 = 100 Hz
+        2 = 200 Hz
+        3 = 400 Hz
+        4 = 800 Hz
+        5 = 1000 Hz
+        6 = 1600 Hz
+        7 = 3200 Hz
+        Default: 0, same as after Power Reset
+    led-slot1:
+      type: int
+      required: false
+      default: 0
+      enum:
+       - 0
+       - 1
+       - 2
+       - 3
+       - 4
+       - 5
+       - 6
+      description: |
+        Set which LED and pulse amplitude are active in time slot 1.
+        0 = None (disabled)
+        1 = LED1 (red), LED1_PA
+        2 = LED2 (IR), LED2_PA
+        3 = LED3 (green), LED3_PA
+        4 = None (disabled)
+        5 = LED1 (red), PILOT_PA
+        6 = LED2 (IR), PILOT_PA
+        7 = LED3 (green), PILOT_PA
+        Default: 1, as needed by heart rate mode
+    led-slot2:
+      type: int
+      required: false
+      default: 0
+      enum:
+       - 0
+       - 1
+       - 2
+       - 3
+       - 4
+       - 5
+       - 6
+      description: |
+        Set which LED and pulse amplitude are active in time slot 2.
+        0 = None (disabled)
+        1 = LED1 (red), LED1_PA
+        2 = LED2 (IR), LED2_PA
+        3 = LED3 (green), LED3_PA
+        4 = None (disabled)
+        5 = LED1 (red), PILOT_PA
+        6 = LED2 (IR), PILOT_PA
+        7 = LED3 (green), PILOT_PA
+        Default: 1, as needed by SpO2 mode
+    led-slot3:
+      type: int
+      required: false
+      default: 0
+      enum:
+       - 0
+       - 1
+       - 2
+       - 3
+       - 4
+       - 5
+       - 6
+      description: |
+        Set which LED and pulse amplitude are active in time slot 3.
+        0 = None (disabled)
+        1 = LED1 (red), LED1_PA
+        2 = LED2 (IR), LED2_PA
+        3 = LED3 (green), LED3_PA
+        4 = None (disabled)
+        5 = LED1 (red), PILOT_PA
+        6 = LED2 (IR), PILOT_PA
+        7 = LED3 (green), PILOT_PA
+        Default: 0, needs configuration when needed in multi-led mode
+    led-slot4:
+      type: int
+      required: false
+      default: 0
+      enum:
+       - 0
+       - 1
+       - 2
+       - 3
+       - 4
+       - 5
+       - 6
+      description: |
+        Set which LED and pulse amplitude are active in time slot 4.
+        0 = None (disabled)
+        1 = LED1 (red), LED1_PA
+        2 = LED2 (IR), LED2_PA
+        3 = LED3 (green), LED3_PA
+        4 = None (disabled)
+        5 = LED1 (red), PILOT_PA
+        6 = LED2 (IR), PILOT_PA
+        7 = LED3 (green), PILOT_PA
```

Comment:
ditto

If you put the comments in the enum cases as suggested, I think that it is a good way also to catch the mismatch error